library/include/hipblas.h

```diff
@@ -398,7 +398,17 @@ typedef enum
     HIPBLAS_STATUS_UNKNOWN           = 11, /**<  back-end returned an unsupported status code */
 } hipblasStatus_t;
 
+/*! \brief Indicates if scalar pointers are on host or device. This is used for scalars alpha and beta and for scalar function return values. */
+typedef enum
+{
+    HIPBLAS_POINTER_MODE_HOST, /**< Scalar values affected by this variable will be located on the host. */
+    HIPBLAS_POINTER_MODE_DEVICE /**<  Scalar values affected by this variable will be located on the device. */
+} hipblasPointerMode_t;
+
 // set the values of enum constants to be the same as those used in cblas
+#ifndef HIPBLAS_SHARED_ENUMS
+#define HIPBLAS_SHARED_ENUMS
+
```

Comment:
I know we discussed this, but I forget the reasoning why we dismissed putting a wrapper around each enum. Then we can just treat all of these enums exactly the same, rather than having some that are different from others.

```cpp
#ifndef HIPBLAS_FILL_MODE_DECLARED
#define HIPBLAS_FILL_MODE_DECLARED
typedef enum
{
    HIPBLAS_FILL_MODE_UPPER = 121, /**<  Upper triangle */
    HIPBLAS_FILL_MODE_LOWER = 122, /**<  Lower triangle */
    HIPBLAS_FILL_MODE_FULL  = 123
} hipblasFillMode_t;
#elif __cplusplus >= 201103L
static_assert("Inconsistent declaration of HIPBLAS_FILL_MODE_UPPER", HIPBLAS_FILL_MODE_UPPER == 121)
static_assert("Inconsistent declaration of HIPBLAS_FILL_MODE_LOWER", HIPBLAS_FILL_MODE_LOWER == 122)
static_assert("Inconsistent declaration of HIPBLAS_FILL_MODE_FULL", HIPBLAS_FILL_MODE_FULL == 123)
#endif
```

I recall there was at least one example in which that wasn't useful, but did we decide it was useless in general? We don't need that kind of granularity for SOLVER, but I wonder if it might nevertheless be simpler if all enums are declared in the exact same way.

Reply:
To be honest, I don't recall either. Thinking on it now, however, I think it would be good for future proofing. If hipBLAS and hipSOLVER introduced a new enum in the future with its own wrapper, we can ensure that the enum won't go undefined because, e.g., an older version of a library defines `HIPBLAS_SHARED_ENUMS` without it. I'll make the change.